k9mail-library/src/test/java/com/fsck/k9/mail/store/imap/ImapFolderTest.java

```diff
@@ -507,6 +538,36 @@ public void getHighestUid() throws Exception {
     }
 
     @Test
+    public void getHighestUid_imapConnectionThrowsNegativesResponse_shouldReturnMinus1() throws Exception {
+        ImapFolder folder = createFolder("Folder");
+        prepareImapFolderForOpen(OPEN_MODE_RW);
+        List<ImapResponse> imapResponses = singletonList(createImapResponse("* SEARCH 42"));
+        when(imapConnection.executeSimpleCommand("UID SEARCH *:*"))
+                .thenThrow(NegativeImapResponseException.class);
+        folder.open(OPEN_MODE_RW);
+
+        long result = folder.getHighestUid();
+
+        assertEquals(-1L, result);
```

Comment:
Our tests usually follow the arrange, act, assert pattern. Rewriting as suggested makes it more difficult to identify the different stages.